test/integration/proxy.go

```diff
@@ -0,0 +1,89 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v12 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"testing"
+	"time"
+)
+
+type TinyProxy struct {
+	pod        *v1.Pod
+	name       string
+	client     v12.PodInterface
+	IP         string
+	PORT       string
+	address    string
+	status     string
+	ready      bool
+	LogChecker *LogCheck
+	t          *testing.T
+}
+
+func (proxy *TinyProxy) create(t *testing.T, clientset *kubernetes.Clientset) error {
+	proxy.t = t
```

Comment:
Do we need to do this? Can't we just pass the ref to `waitUntilReady`? Not sure what is better.....

Reply:
Yes, you're right, thanks for pointing it out. The plan was to use goroutines to save some time, but this can be done even like you say. But it didn't really help in this case, it saved just a few seconds..